app/services/inbound_webhooks/brevo/assign_delivery_status_and_date_base.rb

```diff
@@ -7,7 +7,7 @@ def initialize(webhook_params:, record:)
       end
 
       def call
-        return if @record.delivery_status.in?(record_class::FINAL_DELIVERY_STATUS)
+        return if record_has_a_failed_delivery_status? && !delivered?
```

Comment:
Si on isole ça dans une méthode on comprend mieux ce qu'il se passe là IMO
```suggestion
        return if record_delivery_failed_and_not_now_delivered?
```

Ce qui m'embête quand même c'est que ça laisse à penser que c'est possible dans tous les cas alors que ça ne l'est que pour le soft bounce

Reply:
👌